src/api/app/services/trigger_controller_service/scm_extractor.rb

```diff
@@ -21,36 +21,60 @@ def call
     private
 
     def github_extractor_payload
-      {
+      payload = {
         scm: 'github',
-        commit_sha: @payload.dig(:pull_request, :head, :sha),
-        pr_number: @payload[:number],
-        source_branch: @payload.dig(:pull_request, :head, :ref),
-        target_branch: @payload.dig(:pull_request, :base, :ref),
-        action: @payload[:action], # TODO: Names may differ, maybe we need to find our own naming (defer to service?)
-        source_repository_full_name: @payload.dig(:pull_request, :head, :repo, :full_name),
-        target_repository_full_name: @payload.dig(:pull_request, :base, :repo, :full_name),
         event: @event,
         api_endpoint: github_api_endpoint
       }
+
+      case @event
+      when 'pull_request'
+        payload.merge!({ commit_sha: @payload.dig(:pull_request, :head, :sha),
+                         pr_number: @payload[:number],
+                         source_branch: @payload.dig(:pull_request, :head, :ref),
+                         target_branch: @payload.dig(:pull_request, :base, :ref),
+                         action: @payload[:action], # TODO: Names may differ, maybe we need to find our own naming (defer to service?)
+                         source_repository_full_name: @payload.dig(:pull_request, :head, :repo, :full_name),
+                         target_repository_full_name: @payload.dig(:pull_request, :base, :repo, :full_name) })
+      when 'push'
+        payload.merge!({ commit_sha: @payload[:after],
+                         branch: @payload[:ref].split('/').last,
```

Comment:
branch names can include `/`

Reply:
Yep, I already have code for this, it's not pushed yet though. I'm working on the specs...